modules/opencascade/opencascade/surface_shape.cpp

```diff
@@ -0,0 +1,65 @@
+#include "surface_shape.h"
+
+#include <opencascade/STEPCAFControl_Reader.hxx>
+#include <opencascade/TopExp_Explorer.hxx>
+#include <opencascade/TopoDS.hxx>
+#include <opencascade/BRep_Builder.hxx>
+#include <opencascade/GeomAPI_ProjectPointOnSurf.hxx>
+#include <opencascade/gp_Pnt.hxx>
+
+namespace SPH
+{
+	//=================================================================================================//
+Vecd SurfaceShape::findClosestPoint(const Vecd &input_pnt)
+{
+
+    Vecd closest_pnt;
+    gp_Pnt point1 = EigenToOcct(input_pnt);
+    gp_Pnt point2;
+    Extrema_ExtAlgo Algo = Extrema_ExtAlgo_Tree;
+    point2 = GeomAPI_ProjectPointOnSurf(point1, surface_, Algo);
```

Comment:
seems line 18 and line 20 can be combined as gp_Pnt point2 = GeomAPI_ProjectPointOnSurf(point1, surface_, Algo);